### PR TITLE
impose certain sanity checks on engineconfig settings

### LINKED ISSUE #1181: Ragged paged attention kernel results in weird inference results.
I might be doing something wrong but:

```
uv run scripts/ray/dev_tpu.py --tpu-name=power-scratch execute\
  -- "cd submodules/levanter && uv run --extra=serve --extra=tpu src/levanter/main/inference_repl.py --command=complete --args='What is the square root of 17?' --max_tokens=32 --checkpoint=Qwen/Qwen3-4B-Instruct-2507 --model.type=qwen3 2>&1 | grep -v 'Could not open'"
```

Syncing local changes to dev-tpu-power-scratch...
2025-09-23 16:44:56,629 - INFO - Syncing 1334 files (git-tracked, modified, and important untracked)...
2025-09-23 16:45:01,452 - INFO - File sync completed successfully
Running: cd submodules/levanter && uv run --extra=serve --extra=tpu src/levanter/main/inference_repl.py --command=complete --args='What is the square root of 17?' --max_tokens=32 --checkpoint=Qwen/Qwen3-4B-Instruct-2507 --model.type=qwen3 2>&1 | grep -v 'Could not open'
Loading Qwen/Qwen3-4B-Instruct-2507...
2025-09-23 23:45:11,680 - INFO - Loaded tokenizer with vocab size 151669 from Qwen/Qwen3-4B-Instruct-2507
Loading weights: 100%|██████████| 219/219 [00:02<00:00, 91.70it/s] 
Loading weights: 100%|██████████| 174/174 [00:02<00:00, 73.10it/s]
Loading weights: 100%|██████████| 5/5 [00:00<00:00, 100.49it/s]
2025-09-23 23:45:19,511 - INFO - Resizing model from 151936 to 151669 to match tokenizer vocab size
2025-09-23 23:45:25,661 - INFO - Starting inference context...
2025-09-23 23:45:25,662 - INFO - Inference thread started
2025-09-23 23:45:25,662 - INFO - Batch processing thread started
2025-09-23 23:45:25,662 - INFO - Inference service initialized and ready
✓ Loaded Qwen/Qwen3-4B-Instruct-2507
2025-09-23 23:45:25,670 - INFO - Enqueuing request InferenceRequest(request_id='req_0', prompt_tokens=[3838, 374, 279, 9334, 3704, 315, 220, 16, 22, 30], max_tokens=32, temperature=0.7, stop_tokens=None, seed=None, future=<Future pending>, n_generations=1, original_prompt='What is the square root of 17?', enable_logprobs=False)
2025-09-23 23:45:25,762 - INFO - Max tokens per seq: 64, per batch: 256
2025-09-23 23:45:34,232 - INFO - extract: appended=1 (drained=1) unmapped=0 finished_count=0
2025-09-23 23:45:34,232 - INFO - Initial prefill and extraction took 8.468s
2025-09-23 23:45:39,356 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,356 - INFO - Decode iter: total 5.124s (device 0.000s, host 5.124s, submit 5.079s), fake_submit 0.000s, 0.78 tok/s, 4 new (extract 0.045s, release 0.000s)
2025-09-23 23:45:39,396 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,396 - INFO - Decode iter: total 0.039s (device 0.000s, host 0.039s, submit 0.001s), fake_submit 0.000s, 101.81 tok/s, 4 new (extract 0.038s, release 0.000s)
2025-09-23 23:45:39,435 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,435 - INFO - Decode iter: total 0.039s (device 0.000s, host 0.039s, submit 0.001s), fake_submit 0.000s, 102.09 tok/s, 4 new (extract 0.038s, release 0.000s)
2025-09-23 23:45:39,474 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,474 - INFO - Decode iter: total 0.039s (device 0.000s, host 0.039s, submit 0.001s), fake_submit 0.000s, 102.19 tok/s, 4 new (extract 0.038s, release 0.000s)
2025-09-23 23:45:39,513 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,513 - INFO - Decode iter: total 0.039s (device 0.000s, host 0.039s, submit 0.001s), fake_submit 0.000s, 102.21 tok/s, 4 new (extract 0.038s, release 0.000s)
2025-09-23 23:45:39,552 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,553 - INFO - Decode iter: total 0.039s (device 0.000s, host 0.039s, submit 0.001s), fake_submit 0.000s, 102.20 tok/s, 4 new (extract 0.038s, release 0.000s)
2025-09-23 23:45:39,592 - INFO - extract: appended=4 (drained=4) unmapped=0 finished_count=0
2025-09-23 23:45:39,592 - INFO - Decode iter: total 0.039s (device 0.000s, host 0.039s, submit 0.001s), fake_submit 0.000s, 102.15 tok/s, 4 new (extract 0.038s, release 0.000s)
2025-09-23 23:45:39,624 - INFO - extract: appended=3 (drained=3) unmapped=0 finished_count=1
2025-09-23 23:45:39,624 - INFO - Releasing finished sequences: locals=[1]
2025-09-23 23:45:39,624 - INFO - Decode iter: total 0.032s (device 0.000s, host 0.032s, submit 0.001s), fake_submit 0.000s, 94.08 tok/s, 3 new (extract 0.031s, release 0.000s)
2025-09-23 23:45:39,624 - INFO - Batch generated in 13.86s, 32 tokens, 2.31 tok/s
2025-09-23 23:45:39,624 - INFO - Batch completed in 13.86s, generated 32 tokens
──────────────────────────── Response 1/1 

│  What horsepower perchc 41113618n of the223 value of 126ore length of 133                                                                                                                            

Tokens - Prompt: 10, Completion: 32, Total: 42
2025-09-23 23:45:39,629 - INFO - Shutting down inference context.
2025-09-23 23:45:39,707 - INFO - Inference thread shutting down
2025-09-23 23:45:40,629 - INFO - Batch processing thread shutting down
Shared connection to 35.204.239.159 closed.

### PR DESCRIPTION
**Pull request by dlwh**
fixes #1181 

We could probably figure out ways to work around some of these limits but meh

### CODE REVIEW

File: src/levanter/inference/engine.py [hunk 1]
```diff
@@ -307,6 +323,7 @@ def _prefill_kernel(
     new_tokens, log_probs = hax.vmap(sampler, "position")(logits_at_samples, temps, key=prng_keys)
 
     # Update decode_state (also enqueues into the main decode queue)
+    print(new_tokens.shape)
```
Comment (rjpower) [inline src/levanter/inference/engine.py]: nit, remove

File: src/levanter/inference/engine.py [hunk 2]
```diff
@@ -105,11 +105,27 @@ class InferenceEngineConfig:
     enable_logprobs: bool = False
     """Enable computing logprobs for generated tokens."""
 
+    def __post_init__(self):
+        # this one is only required because of clones. If we really care, we could relax this
+        if self.max_queued_tokens < self.max_seqs:
```
Comment (rjpower) [inline src/levanter/inference/engine.py:110]: I wonder if there's a sensible way to derive some of these from the others so that the user specified values couldn't conflict, e.g. now we have:

```
                max_pages=32,
                max_seqs=2,
                page_size=8,
                max_pages_per_seq=8,
                max_queued_tokens=4,
                max_seqs_in_prefill=2,
                max_queued_tokens=64,
                max_prefill_size=64,
                max_tokens_per_round=16,
                max_rounds=8,
```

would it be reasonable to make some of these multipliers of other constants:

```
max_seqs
max_tokens_per_seq
max_tokens_per_round
page_size
->
max_prefill_size = max_tokens_per_seq
max_pages = max_tokens_per_seq * max_seqs / page_size
max_rounds = max_tokens_per_round ...
```

etc. Not needed for this PR but just a thought.

Comment (dlwh) [reply src/levanter/inference/engine.py:110]: i'll try to clean this up when i do the hbm_utilization stuff
